addons/binding/org.openhab.binding.http/src/main/java/org/openhab/binding/http/internal/util/HttpUtil.java

```diff
@@ -0,0 +1,247 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.http.internal.util;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.api.Response;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.eclipse.smarthome.core.library.types.RawType;
+import org.openhab.binding.http.internal.HttpBindingConstants;
+
+import java.io.ByteArrayOutputStream;
+import java.net.URL;
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Utilities for making HTTP requests.
+ *
+ * @author Brian J. Tarricone - Initial contribution
+ */
+@NonNullByDefault
+public class HttpUtil {
```

Comment:
I think this class is more complicated than it needs to be with the CompletionStateResponseListeners etc. Have a look at my class:
https://github.com/openhab/openhab2-addons/blob/master/addons/binding/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/netutils/AsyncHttpClient.java

Especially line 112. Just use `getContentAsString()` within a `BufferingResponseListener`.

Reply:
Ah, wasn't aware of `BufferingResponseListener`!  Thanks, updating...